cupyx/scipy/signal/windows/_windows.py

```diff
@@ -0,0 +1,2197 @@
+
+    return _truncate(w, needs_trunc)
+
+
+def _fftautocorr(x):
+    """Compute the autocorrelation of a real array and crop the result."""
+    N = x.shape[-1]
+    use_N = cupy.fft.next_fast_len(2 * N - 1)
+    x_fft = cupy.fft.rfft(x, use_N, axis=-1)
+    cxy = cupy.fft.irfft(x_fft * x_fft.conj(), n=use_N)[:, :N]
+    # Or equivalently (but in most cases slower):
+    # cxy = np.array([np.convolve(xx, yy[::-1], mode='full')
+    #                 for xx, yy in zip(x, x)])[:, N-1:2*N-1]
+    return cxy
+
+
+_win_equiv_raw = {
+    ("barthann", "brthan", "bth"): (barthann, False),
+    ("bartlett", "bart", "brt"): (bartlett, False),
+    ("blackman", "black", "blk"): (blackman, False),
+    ("blackmanharris", "blackharr", "bkh"): (blackmanharris, False),
+    ("bohman", "bman", "bmn"): (bohman, False),
+    ("boxcar", "box", "ones", "rect", "rectangular"): (boxcar, False),
+    ("chebwin", "cheb"): (chebwin, True),
+    ("cosine", "halfcosine"): (cosine, False),
+    ("exponential", "poisson"): (exponential, True),
+    ("flattop", "flat", "flt"): (flattop, False),
+    ('general cosine', 'general_cosine'): (general_cosine, True),
+    ("gaussian", "gauss", "gss"): (gaussian, True),
+    (
+        "general gaussian",
+        "general_gaussian",
+        "general gauss",
+        "general_gauss",
+        "ggs",
+    ): (general_gaussian, True),
+    ('general hamming', 'general_hamming'): (general_hamming, True),
+    ("hamming", "hamm", "ham"): (hamming, False),
+    ("hanning", "hann", "han"): (hann, False),
+    ("kaiser", "ksr"): (kaiser, True),
+    ("nuttall", "nutl", "nut"): (nuttall, False),
+    ("parzen", "parz", "par"): (parzen, False),
+    # ('slepian', 'slep', 'optimal', 'dpss', 'dss'): (slepian, True),
+    ("triangle", "triang", "tri"): (triang, False),
+    ("tukey", "tuk"): (tukey, True),
+}
+
+# Fill dict with all valid window name strings
+_win_equiv = {}
+for k, v in _win_equiv_raw.items():
+    for key in k:
+        _win_equiv[key] = v[0]
+
+# Keep track of which windows need additional parameters
+_needs_param: Set[str] = set()
+for k, v in _win_equiv_raw.items():
+    if v[1]:
+        _needs_param.update(k)
+
+
+def get_window(window, Nx, fftbins=True):
+    r"""
+    Return a window of a given length and type.
+
+    Parameters
+    ----------
+    window : string, float, or tuple
+        The type of window to create. See below for more details.
+    Nx : int
+        The number of samples in the window.
+    fftbins : bool, optional
+        If True (default), create a "periodic" window, ready to use with
+        `ifftshift` and be multiplied by the result of an FFT (see also
+        `fftpack.fftfreq`).
+        If False, create a "symmetric" window, for use in filter design.
+
+    Returns
+    -------
+    get_window : ndarray
+        Returns a window of length `Nx` and type `window`
+
+    Notes
+    -----
+    Window types:
+
+    - `~cupyx.scipy.signal.windows.windows.boxcar`
```

Comment:
Seems the document is broken.
```suggestion
    - `~cupyx.scipy.signal.windows.boxcar`
```